.github/workflows/build-and-push-image.yml

```diff
@@ -91,3 +91,24 @@ jobs:
             } catch (error) {
               core.setFailed(error.message);
             }
+
+      - name: Trigger Build and Push NuGet Package workflow
+        uses: "actions/github-script@v7"
+        with:
+            github-token: "${{ secrets.GITHUB_TOKEN }}"
+            script: |
+              const tagName = ${{ needs.set-env.outputs.release }}
+              if (!tagName.startsWith("production-")) {
+                console.log("Not a production deployment. Skipping trigger.");
+                return;
+              }
```

Comment:
you can check the `environment` is `production` as an `if` on the job step instead of checking the tag name